samcli/lib/docker/log_streamer.py

```diff
@@ -0,0 +1,70 @@
+"""
+Log streaming utilities when streaming logs from Docker
+"""
+import os
+
+from samcli.lib.package.stream_cursor_utils import cursor_up, cursor_left, cursor_down, clear_line
+
+
+class LogStreamer:
+    def __init__(self, stream, error_class, error_msg_prefix=""):
+        self.stream = stream
+        self.error_class = error_class
+        self.error_msg_prefix = error_msg_prefix
+
+    def stream_progress(self, logs):
+        """
+        Stream progress from docker push logs and move the cursor based on the log id.
+        :param logs: generator from docker_clent.api.push
+        """
+        ids = dict()
+        for log in logs:
+            _id = log.get("id", "")
+            status = log.get("status", "")
+            stream = log.get("stream", "")
+            progress = log.get("progress", "")
+            error = log.get("error", "")
+            change_cursor_count = 0
+            if _id:
+                try:
+                    curr_log_line_id = ids[_id]
+                    change_cursor_count = len(ids) - curr_log_line_id
+                    self.stream.write((cursor_up(change_cursor_count) + cursor_left).encode())
```

Comment:
With the amount of self.stream.write(string.encode()) maybe we can separate this out into its own method which is easier to read.